### PR TITLE
Sarif output structure update.

### LINKED ISSUE #28: SARIF output structure addition
### Feature Description

The consumers want more information listed about the security alert and rules. We can bring these information from the JSON fetched. So the output file needs a structure add-ons. 

### Problem / Opportunity

Opportunity to have full output with all wanted information.

### Acceptance Criteria

Description of alert should contain:
Avd_id:
Repository_full_name:
Reachable:
Scan_date:
First_seen:
Scm_file:
Target_line:
Istalled_version:

Rule should contain:
Category: sast/vulnerabilities/iacMis
Avd_id:
Title:
Severity:
Fixed_version:
Published_date:
Package_name:
ExtraData.Owasp:
ExtraData.Category
ExtraData.Impact:
ExtraData.Confidence:
ExtraData.Likelihood:
ExtraData.Remediation:
ExtraData.References:

### Proposed Solution

Use the JSON fetched and add the data into SARIF converted output.

### Dependencies / Related

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by tmikula-dev**
## Overview
This pull request significantly refactors and enhances the SARIF conversion logic in `src/model/sarif_convertor.py`, focusing on improving the formatting and completeness of help and message texts in SARIF output.

## Release Notes
- Update of the Sarif structure output

## Related
Closes #28 

## Summary by CodeRabbit

* **Refactor**
  * Improved SARIF output: rule help and finding messages are now assembled from structured fields, include optional OWASP and reference sections, and consistently include file location when available.

* **Tests**
  * Expanded coverage for SARIF generation validating comprehensive/omitted fields, list rendering, message formatting, rule deduplication, metadata handling, and edge-case line/format behaviors.

### CODE REVIEW

File: tests/model/test_sarif_convertor.py [hunk 1]
```diff
@@ -159,6 +159,186 @@ def test_convert_to_sarif_includes_reference_as_help_uri():
...
+    assert "**Fixed version:**" not in help_text
+    assert "**Published date:**" not in help_text
+    assert "**Package name:**" not in help_text
+    assert "**OWASP:**" not in help_text
+    assert "**Category:**" not in help_text
+    assert "**Impact:**" not in help_text
+    assert "**Confidence:**" not in help_text
+    assert "**Likelihood:**" not in help_text
+    assert "**Remediation:**" not in help_text
+    assert "**References:**" not in help_text
```
[the remark this answers is not in the document]
Comment (tmikula-dev) [reply tests/model/test_sarif_convertor.py]: These parts were put out from the criteria, they will not be added. So the test is correct.

File: src/model/sarif_convertor.py [hunk 2]
```diff
@@ -218,39 +232,122 @@ def _build_sarif_finding(self, finding_json: Finding, rule_index: int) -> dict:
...
+        start_line = finding_json.get("target_start_line", 0)
+        end_line = finding_json.get("target_end_line", 0)
+
+        if start_line and isinstance(start_line, int) and start_line > 0:
+            region: dict[str, Any] = {
+                "startLine": start_line,
+                "startColumn": 1,
             }
+            if end_line and isinstance(end_line, int) and end_line > 0 and end_line != start_line:
+                region["endLine"] = end_line
```
[the remark this answers is not in the document]
Comment (tmikula-dev) [reply src/model/sarif_convertor.py]: [f9dbfdf](https://github.com/AbsaOSS/aquasec-scan-results/pull/53/commits/f9dbfdf429412d113941af61f1cb505f6fbc305c)
